contracts/Vault.vy

```diff
@@ -581,6 +581,14 @@ def setWithdrawalQueue(queue: address[MAXIMUM_STRATEGIES]):
         if queue[i] == ZERO_ADDRESS and self.withdrawalQueue[i] == ZERO_ADDRESS:
             break
```

Comment:
Just noticed `self.withdrawalQueue[i] == ZERO_ADDRESS` is needed to break out of this loop.... Thinking about this condition, it doesn't seem to make sense and could be a weird corner case to this logic. I would suggest that instead it is modified to this:

```suggestion
        # old_queue[i] = self.withdrawalQueue[i]  # `old_queue` is defined before loop
        if queue[i] == ZERO_ADDRESS:
            # NOTE: Cannot use this method to remove entries from the queue
            assert old_queue[i] == ZERO_ADDRESS
            break
        # NOTE: Cannot use this method to add more entries to the queue
        assert old_queue[i] != ZERO_ADDRESS
```

---

Side note: we should have better testing around this logic to show that we can't add or remove entries from the queue using this reordering method. With the above suggested changes, we would have a check that entries are neither added nor removed, and that there are no duplicates. However, a forth condition is needed, and that is "all items in queue prior to reordering are still in queue afterwards".

Reply:
Got it to work with a check to make sure a new strategy isn't replacing an older one.